apps/aecore/src/aec_block_fork.erl

```diff
@@ -36,3 +37,152 @@ apply_accounts_file(Trees, Accounts) ->
             Accounts),
     aec_trees:set_accounts(Trees, AccTrees).
 
+%%% ===========================================
+%%% Inserting a contract at hard fork height
+%%% ===========================================
+%%%
+%%%  This is introduced at Lima hard fork to enable distributing funds
+%%%  to accounts that missed the final migration deadline.
+%%%
+%%%  The contracts are created in the normal way, using a
+%%%  contract_create_tx, but to avoid side effects from the init call,
+%%%  the transactions are executed in isolation on the same state
+%%%  trees (i.e., the state trees coming in to the key block).
+%%%
+%%%  The new contracts objects and their accounts are then inserted
+%%%  into the original state trees, one by one.
+%%%
+%%%  The owner of the contracts (i.e., the originator of the fictive
+%%%  create transactions) is the lock account from governance.  Since
+%%%  the contract pubkeys depends on the nonce of this account, the
+%%%  nonce must be bumped, but the balance of this account is unchanged.
+%%%
+%%%  The end result is that the state trees is transformed as:
+%%%    - The new contracts are inserted
+%%%    - The new contracts have balances from freshly created tokens
+%%%    - The lock account will have a new nonce.
+%%%
+%%%  Note that there are no call objects, and no transactions recorded on
+%%%  the chain, and no fees or gas compensation, is given for this execution.
+%%%
+%%%  Also note that any side effects of executing the init function
+%%%  will only affect the account itself. It is therefore highly
+%%%  discouraged to have other side effects in the init call.
+
+
+apply_hard_fork_contracts_file([], Trees,_TxEnv) ->
+    %% Nothing to do
+    Trees;
+apply_hard_fork_contracts_file(Specs0, Trees, TxEnv) ->
+    Specs         = sort_by_nonce(Specs0),
+    Static        = fork_contracts_static_specs(TxEnv),
+    OwnerPubkey   = maps:get(owner_pubkey, Static),
+    CreateTxs     = fun(Spec) -> contract_create_tx(Spec, Static, TxEnv) end,
+    Txs           = lists:map(CreateTxs, Specs),
+    Trees1        = fork_contracts_prepare_owner(Txs, Specs, Static, Trees),
+    ApplyTxs      = fun(Tx, T) -> apply_hard_fork_contract_tx(Tx, T, TxEnv) end,
+    {Contracts, Trees2} = lists:mapfoldl(ApplyTxs, Trees1, Txs),
+    Nonce         = get_nonce(OwnerPubkey, Trees2),
+    hard_fork_contracts_post_processing(Contracts, OwnerPubkey, Nonce, Trees).
+
+fork_contracts_static_specs(TxEnv) ->
+    OwnerPubkey   = aec_governance:locked_coins_holder_account(),
+    GasLimit      = aec_governance:block_gas_limit(),
+    GasPrice      = aec_governance:minimum_gas_price(aetx_env:height(TxEnv)),
```

Comment:
OK, since we don't go through the mempool. (I was worried about the miner minimum gas price for a short while.)